src/socketio.quiver.js

```diff
@@ -287,24 +366,26 @@ QuiverSocialProvider.prototype.connectLoop_ = function(servers, connector) {
     reject = R;
   });
 
-  var serverKeys = Object.keys(servers);
-  var tried = 0, connected = 0;
-  var helper = function(success, failure) {
-    if (success) {
-      ++connected;
-    }
-    if (connected === QuiverSocialProvider.MAX_CONNECTIONS_ ||
-        (tried === serverKeys.length && connected > 0)) {
-      fulfill();
-      return;
+  var serverKeys = QuiverSocialProvider.shuffle_(Object.keys(servers));
+  var tried = 0;
+  /** @type {!Array<!QuiverSocialProvider.server_>} */
+  var succeeded = [];
+  /** @type {!Array<!QuiverSocialProvider.server_>} */
+  var failed = [];
+  var helper = function(retval, failure) {
+    var lastServer = servers[serverKeys[tried - 1]];
+    if (failure) {
+      failed.push(lastServer);
+    } else {
```

Comment:
Er, I just realized ```success``` is  not defined here.  But somehow this should be fixed so that ```undefined``` is not pushed onto the succeeded array on the first call, maybe check for ```tried > 0```